### PR TITLE
Add node config to TaskInstance Context

### PR DESCRIPTION
**Pull request by linchun3**
## Description

Add the node's attributes (config, tags, etc, ...) into a TaskInstance context for retrieval by callback functions in Airflow through the use of `pre_execute` to store these attributes into a task's context.

As [this PR](https://github.com/astronomer/astronomer-cosmos/pull/700/files) seems to be closed, and I have a use case for this feature, I attempt to recreate the needed feature. 

We leverage the `context_merge` utility function from Airflow to merge the extra context into the `Context` object of a `TaskInstance`.

## Related Issue(s)
closes https://github.com/astronomer/astronomer-cosmos/issues/698

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cosmos/airflow/graph.py [hunk 1]
```diff
@@ -102,6 +104,9 @@ def create_test_task_metadata(
             task_args["select"] = f"source:{node.resource_name}"
         else:  # tested with node.resource_type == DbtResourceType.SEED or DbtResourceType.SNAPSHOT
             task_args["select"] = node.resource_name
+
+        extra_context = {f"{node.resource_type.value}_config": node.config}
```
Comment (tatiana) [inline cosmos/airflow/graph.py:108]: I'm pretty curious here, why did you add the dbt node type in the key of the config, as opposed to something like `cosmos_node_config`?

**Review by tatiana — COMMENTED** [review]
Hi @linchun3 I left a minor question - we're close to merging this and I think we will be able to ship it in 1.5.

File: cosmos/airflow/graph.py (hunk 1 above)
Comment (linchun3) [reply cosmos/airflow/graph.py:108]: > Since this is per task/node, wouldn't it be simpler always to have the same context identifier - and if the user is interested in the node type, they can check on the node.config

@tatiana oh yeah this makes sense, there would be less context switching required for the user using the field in the callback too. I'll make the change :) 

> Also, wouldn't it be more useful if we saved other properties of the dbt node to the context, including tags and resource type..? Perhaps we could have a property in DbtNode called context_dict to expose all the information of interest.

It'll be useful to expose all the attributes of a node, should I add that too in this PR?

Comment (linchun3) [reply cosmos/airflow/graph.py]: I have added the entire `DbtNode` object as a dictionary into `dbt_node_config` key as `extra_context`

**Review by tatiana — APPROVED** [review]
Thanks a lot for implementing this and addressing the feedback, @linchun3 ! It looks great!
